pulp_python/app/tasks/publish.py

```diff
@@ -93,7 +94,7 @@ def write_simple_api(publication):
         .distinct()
     )
 
-    index_names = [(name, sanitize_name(name)) for name in project_names]
+    index_names = [(name, canonicalize_name(name)) for name in project_names]
```

Comment:
I like de-duplication and using canonicalize_name found in packaging.utils instead of importing re and writing one ourselves, but correct me if I'm wrong (my regex is a bit rusty) the sanitize_name stripped out *all* non-alphanumeric characters whereas canonicalize_name only strips out '-', '_', and '.'.  Can we assume we won't encounter any other characters for our purposes?  I know there aren't supposed to be for naming conventions but does PYPI and anything else we may use enforce this?

Reply:
I checked the source code when changing this and it's actually the exact same regex as the current one. ```packaging``` is an official PyPA package and so it makes sense to just use their function instead of having our own copy.